lib/hooks/userspace/hook-hapi.js

```diff
@@ -88,7 +89,7 @@ function startRootSpanForRequest(req, traceHeader) {
 
   // we use the path part of the url as the span name and add the full
   // url as a label
-  var rootContext = agent.createRootSpanData(req.url, traceId, parentSpanId, 3);
+  var rootContext = agent.createRootSpanData(urlParse(req.url).pathname, traceId, parentSpanId, 3);
```

Comment:
Wouldn't `req.path` work here?

Doc: http://hapijs.com/api#request-properties
Available in hapi@8 as well as hapi@latest as per the docs.